modules/oe_theme_helper/src/TwigExtension/Filters.php

```diff
@@ -73,14 +75,35 @@ public function toNativeLanguageName($language_code): string {
       return $languages[$language_code]->getName();
     }
     // The fallback implemented in case we don't have enabled language.
-    $predefined = self::getEuropeanUnionLanguageList() + LanguageManager::getStandardLanguageList();
+    $predefined = EuropeanUnionLanguages::getLanguageList() + LanguageManager::getStandardLanguageList();
     if (!empty($predefined[$language_code][1])) {
       return $predefined[$language_code][1];
     }
 
     throw new \InvalidArgumentException('The language code ' . $language_code . ' does not exist.');
   }
 
+  /**
+   * Get an internal language ID given its code.
+   *
+   * @param string $language_code
+   *   The language code as defined by the W3C language tags document.
+   *
+   * @return string
+   *   The internal language ID.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when the passed in language code does not exist.
+   */
+  public function toInternalLanguageId($language_code): string {
+    // The fallback implemented in case we don't have enabled language.
+    if (EuropeanUnionLanguages::hasLanguage($language_code)) {
+      return EuropeanUnionLanguages::getInternalLanguageCode($language_code);
+    }
+
+    throw new \InvalidArgumentException('The language code ' . $language_code . ' does not exist.');
```

Comment:
This exception is misleading. The language code is not in the European Union language list but it could have been enabled in the site (lets say chinese for example). I think we should check for the internal code language in the language list but return the language code if it is not found. (I know this is an scenario that will probably never occur and atm we only use this method to pick the appropriate logo, but we can rule out it could be used somewhere else)

Reply:
That's a good point, I'll change that.